CMakeLists ignores important find_package variables

<!--
  Fill in the placeholders below. Delete any headings and placeholders that you do not fill in.
-->
**OS:** Xubuntu 16.04
**Version:** 0.2.0

<!-- Explanation of the issue -->

Compilation error:
```
/home/simmo/ORCT2/OpenRCT2/src/openrct2/core/String.cpp: In function ‘std::__cxx11::string String::CodePageFromUnicode(icu_55::UnicodeString, sint32)’:
/home/simmo/ORCT2/OpenRCT2/src/openrct2/core/String.cpp:647:57: error: invalid conversion from ‘const UChar*’ {aka ‘const short unsigned int*’} to ‘const char16_t*’ [-fpermissive]
         const char16_t* source = src.getTerminatedBuffer();
                                  ~~~~~~~~~~~~~~~~~~~~~~~^~
/home/simmo/ORCT2/OpenRCT2/src/openrct2/core/String.cpp:652:87: error: invalid conversion from ‘const char16_t*’ to ‘const UChar*’ {aka ‘const short unsigned int*’} [-fpermissive]
         ucnv_fromUnicode(conv, &buffer_target, buffer_limit, (const UChar**) &source, source_limit, nullptr, true, &status);
```

---

After some tinkering I've realized that OpenRCT2 uses `find_package` in CMake incorrectly. For example here: https://github.com/OpenRCT2/OpenRCT2/blob/66bfd425517f2ea5408902a99e44bb0c06767792/src/openrct2/CMakeLists.txt#L115-L120

`find_package` is used only for checking dependency versions and in this case also `target_link_libraries`. The problem is that `find_package` scripts set many other variables also which need to be explicitly used for the found package to correctly work. In my case there needs to be `include_directories(${ICU_INCLUDE_DIRS})` in there as well. Not having that makes compilation still use system installed libraries (e.g. includes) instead of the ones configured in CMake. This is extremely misleading since CMake shows the correct paths while the compilation actually doesn't since OpenRCT2's CMake script never added them to the compilation.

There may be similar issues with other dependencies where necessary `find_package` information is just ignored but should not be.

I could for the ICU includes. There's another issue with that too, not sure what should be done about it: ICU headers don't use `override` so including them causes errors like
```
/home/simmo/ORCT2/icu-root/include/unicode/bytestream.h:174:16: error: ‘virtual void icu_62::CheckedArrayByteSink::Append(const char*, int32_t)’ can be marked override [-Werror=suggest-override]
   virtual void Append(const char* bytes, int32_t n);
```

Personally I just put `-Wno-error=suggest-override` into CXX_FLAGS but it doesn't seem like a good solution. I'm wondering how this hasn't come up for anyone before? Is it some fresh g++/CMake addition to add those warnings or what?

--- a/src/openrct2/CMakeLists.txt
+++ b/src/openrct2/CMakeLists.txt
@@ -162,6 +162,9 @@ if (NOT DISABLE_TTF)
         target_include_directories(${PROJECT} PRIVATE ${FONTCONFIG_INCLUDE_DIRS})
     endif ()
 endif ()
+if (NOT MINGW AND NOT MSVC)
+    target_include_directories(${PROJECT} SYSTEM PRIVATE ${ICU_INCLUDE_DIRS})
+endif ()
 
 # To avoid unnecessary rebuilds set the current branch and
 # short sha1 only for the two files that use these

